### ISSUE TITLE
Importing patients from old platform

### ISSUE THREAD

**Issue by AmaniKrayemRCPCH** [rcpch/rcpch-audit-engine#596]
Headers - [Export_Final.csv](https://github.com/rcpch/rcpch-audit-engine/files/12867570/Export_Final.csv)
Data format - [Key_Final.csv](https://github.com/rcpch/rcpch-audit-engine/files/12867572/Key_Final.csv)
50 dummy patients - [DummyPatients_Final.csv](https://github.com/rcpch/rcpch-audit-engine/files/12867581/DummyPatients_Final.csv)

A couple notes:
- Each patient is assigned to Trust. The referring hospital column is free-text and not mandatory.
- I've also kept the ethnicity other free-text column, but it may not actually be of use here. 
- There are some Trust mergers that aren't reflected in our old platform. The details of mergers are here - [Trust Mergers for Import.xlsx](https://github.com/rcpch/rcpch-audit-engine/files/12868172/Trust.Mergers.for.Import.xlsx) This is not comprehensive as we only note the mergers that we have been informed of.

**Comment by AmaniKrayemRCPCH** [comment]
Another thing to note is that NHS numbers and postcodes aren't validated on the old platform. I've put valid data in the dummy patients CSV from online generators, but you will probably get errors when importing the actual data.

**Comment by anchit-chandran** [comment]
Hey @AmaniKrayemRCPCH , just to confirm - how does the `S01DOBDateOnly` col correspond to the defined `S01DOB (datetime)` header?




![Image](https://github.com/rcpch/rcpch-audit-engine/assets/68898006/48da1e23-dbee-4011-9fd7-a98b8e3ed65e)

**Comment by anchit-chandran** [comment]
Also wanted to confirm, went through the different available options from your data for ethnicities and mapped to what made most sense on how we define them.

Pls could you let me know if this is fine / any changes required? The format is:

`netsolveEthnicity ~> ourEthnicity (description)`

```
WBritish ~> A (British, Mixed British)
WIrish ~> C (Any other White background)
WOther ~> C (Any other White background)
BCaribbean ~> M (Caribbean)
BAfrican ~> N (African)
BOther ~> P (Any other Black background)
AIndian ~> H (Indian or British Indian)
APakistani ~> J (Pakistani or British Pakistani)
ABangladeshi ~> K (Bangladeshi or British Bangladeshi)
AOther ~> L (Any other Asian background)
MWBC ~> D (Mixed (White and Black Caribbean))
MWBA ~> E (Mixed (White and Black African))
MWAs ~> F (Mixed (White and Asian))
MOther ~> G (Any other mixed background)
OChinese ~> R (Chinese)
OOther ~> S (Any other ethnic group)
No ~> Z (Not Stated)
NK ~> Z (Not Stated)
```

**Comment by anchit-chandran** [comment]
Query 3 sorry:

we don't have any organisations for these 3 Trusts. Because each Case must be registered to an Organisation, I'm unsure what to do with these records.




![Image](https://github.com/rcpch/rcpch-audit-engine/assets/68898006/733a47a6-2900-4145-b8c4-7a67b13f4901)

**Comment by AmaniKrayemRCPCH** [comment]
Hi Anchit,

City Hospitals Sunderland NHS Foundation Trust (RLN) and South Tyneside NHS Foundation Trust (RE9) have merged, and are now South Tyneside and Sunderland NHS Foundation Trust (R0B).

Organisations:
R0B01 - Sunderland Royal Hospital, Kayll Road, Sunderland, SR4 7TP.


North Cumbria University Hospitals NHS Trust (RNL) merged with Cumbria Partnership NHS Foundation Trust to create North Cumbria Integrated Care NHS Foundation Trust (RNN).

Organisations:
RNN62 – Cumberland Infirmary, Newtown Road, Carlisle, Cumbria, CA2 7HY
RNNBX – West Cumberland Hospital, Homewood, Hensingham, Whitehaven, Cumbria, CA28 8JG




Amani Krayem (she/her)
Epilepsy12 Project Coordinator
Royal College of Paediatrics and Child Health
5-11 Theobalds Road, London, WC1X 8SH
020 7092 6157
www.rcpch.ac.uk/epilepsy12<http://www.rcpch.ac.uk/epilepsy12>
***@***.***@epilepsy_12

Leading the Way in Children's Health
The Royal College of Paediatrics and Child Health (RCPCH) is a registered charity in England and Wales (1057744) and in Scotland (SC038299)

**Comment by AmaniKrayemRCPCH** [comment]
Heya,


I think it should be

WIrish ~> B (Irish)

Otherwise, ethnicities are all correct





Amani Krayem (she/her)
Epilepsy12 Project Coordinator
Royal College of Paediatrics and Child Health
5-11 Theobalds Road, London, WC1X 8SH
020 7092 6157
www.rcpch.ac.uk/epilepsy12<http://www.rcpch.ac.uk/epilepsy12>
***@***.***@epilepsy_12

Leading the Way in Children's Health
The Royal College of Paediatrics and Child Health (RCPCH) is a registered charity in England and Wales (1057744) and in Scotland (SC038299)

**Comment by AmaniKrayemRCPCH** [comment]
In regards to the DOB column, that was an excel formatting issue. 
Here's the fixed dataset - 
[DummyPatients_Final_20231020.csv](https://github.com/rcpch/rcpch-audit-engine/files/13054312/DummyPatients_Final_20231020.csv)

**Comment by ollyrice** [comment]
I'll be creating a SharePoint site "E12 data transfer" as a secure location to upload of exported NetSolving data.
Olly

**Comment by nikyraja** [comment]
Thanks Olly for creating the SP site. The patient data has now been uploaded onto this site. 
- A note that the columns are in a slightly different order to the dummy patient version where DOB is now the final column. If this is an issue and we can redo. 

This is ready for importing into the new platform - please let us know if you need anything!

**Comment by nikyraja** [comment]
A note from Hannah on the download:

'For sitecode RH5 the sitename should be Somerset NHS Foundation Trust not Yeovil District Hospital NHS Foundation Trust. 
Also, where it says NHS in the sitename and s01refferringhospital it will be written as Nhs rather than NHS.'

An updated version had been uploaded onto the site: 'Epilepsy12_Cohort6_APIexport_reformated_151123'

**Comment by pacharanero** [comment]
WIP on this is in the #635 pull request.
Note that a previous Issue #552 suggesting the use of REST API for this has been superseded in favour of a simpler CSV importer.

**Comment by pacharanero** [comment]
@AmaniKrayemRCPCH @nikyraja the initial import of patient data is now done on Staging, where you can log in to review and spot-check some cases, just to ensure we've got it all right. If Staging is OK then we do the same on Live and you are ready to go. (After which I'll delete data on Staging and revert to some dummy data)

**Comment by nikyraja** [comment]
Thanks @pacharanero and all for this!

We have reviewed the data and performed random spot checks. I am satisfied that all the data that has been imported into the new platform is accurate against the data we supplied and the current NetSolving platform.
- I am happy for you to move this to the live site and delete the data off staging. 

A file has been uploaded to the SP site which describes our spot checking process/results. There were a few patients which didn't import, possible due to incorrect postcodes. As discussed over email, we are happy for you to provide the details of these patients (as this is a small number) and we can enter these manually or inform the clinical team for them to register directly. 

We also could not find any patients on the new platform under COX3P (Manchester local care organisation), despite there being ~20 CYP within the data we supplied. 

I am still happy to proceed despite the above as all the data on the new platform is accurate, and as long as we have a list of 'failed imports' we can communicate this to teams. 

This was a big task, thank you for completing this so accurately & quickly!
